swift/log/SwiftDiagnostics.h

```diff
@@ -83,4 +83,11 @@ inline void internal_error() {
 }
 }  // namespace diagnostics
 
+namespace detail {
+template <void (*Func)()>
+inline void createSwiftDiagnosticsSourceOnce() {
+  static int ignore = (Func(), 0);
```

Comment:
Can we just use `std::call_once` instead?

Nit: `std::ignore` is only specified for use with `std::tie`. I'd expect it to work in practice, but if the goal is just to silence an unused-variable warning, casting to void is a conforming alternative.

Reply:
yeah, `std::call_once` is a better solution.

I've always found the `std::ignore = x` idiom nicer to read than the cast to void though